auth/src/main/kotlin/com/hedvig/android/auth/network/SubmitAuthorizationCodeResponse.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.android.auth.network
+
+import com.hedvig.android.auth.AccessToken
+import com.hedvig.android.auth.AuthTokenResult
+import com.hedvig.android.auth.RefreshToken
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Response
+
+@Serializable
+data class SubmitAuthorizationCodeResponse(
+  @SerialName("access_token")
+  val accessToken: String,
+  @SerialName("expires_in")
+  val accessTokenExpiresIn: Int,
+  @SerialName("refresh_token")
+  val refreshToken: String,
+  @SerialName("refresh_token_expires_in")
+  val refreshTokenExpiresIn: Int,
+)
+
+fun Response.toAuthTokenResult(): AuthTokenResult {
+  val responseBody = body?.string()
+  val result = if (isSuccessful && responseBody != null) {
+    Json.decodeFromString<SubmitAuthorizationCodeResponse>(responseBody).toAuthAttemptResult()
```

Comment:
One thing I thought of for this, we use the global `Json` in all these samples. Is the default configuration enough? In Odyssey we had to make one ourselves for a couple of reasons if I remember correctly. Will just have to double check if we want to do that here too ☺️